src/현정빈/Myclass.java

```diff
@@ -0,0 +1,148 @@
+        }
+    }
+
+    /**
+     * 저장소에서 멤버를 삭제하는 메서드,
+     * 존재하지 않는 id라면 false를 반환
+     */
+    @Override
+    public boolean removeMember(int id) {
+        if(!map.containsKey(id)){//지정된 id를 포함하고 있는지 여부를 반환, 반환값이 false라면
+            return false;
+        }
+        else {
+            map.remove(id); //지정된 id와 연결된 값을 map에서 제거
+            return true;
+        }
+    }
+
+    /**
+     * 멤버를 수정하는 메서드 1,
+     * id를 기준으로 멤버를 찾아서, 이름을 수정함,
+     * 존재하지 않는 id라면 false를 반환
+     */
+    @Override
+    public boolean updateMember(int id, String name) {
+        if(map.containsKey(id)) {//지정된 id를 포함하고 있는지 여부를 반환, 반환값이 true라면
+            map.get(id).setName(name);//map에서 지정된 id에 연결된 값(참조값=주소)을 반환하면서 Member의 메소드를 사용할 수 있게 된다
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * 멤버를 수정하는 메서드 2,
+     * id를 기준으로 멤버를 찾아서 나이를 수정함,
+     * 존재하지 않는 id라면 false를 반환
+     */
+    @Override
+    public boolean updateMember(int id, int age) {
+        if(map.containsKey(id)) {//지정된 id를 포함하고 있는지 여부를 반환, 반환값이 true라면
+            map.get(id).setAge(age);//map에서 지정된 id에 연결된 값(참조값=주소)을 반환하면서 Member의 메소드를 사용할 수 있게 된다
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * 멤버를 수정하는 메서드 3,
+     * id를 기준으로 멤버를 찾아서 이름과 나이를 수정함,
+     * 존재하지 않는 id라면 false를 반환
+     */
+    @Override
+    public boolean updateMember(int id, String name, int age) {
+        if(map.containsKey(id)) { //지정된 id를 포함하고 있는지 여부를 반환, 반환값이 true라면
+            map.get(id).setName(name); //map에서 지정된 id에 연결된 값(참조값=주소)을 반환하면서 Member의 메소드를 사용할 수 있게 된다
+            map.get(id).setAge(age);
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+
+    /**
+     * id를 통해 멤버를 검색하는 메서드
+     */
+    @Override
+    public Member findById(int id) {
+        if(map.containsKey(id)){ //맵이 지정된 id를 포함하고 있는지 여부 반환, 반환된 값이 true라면
+            return map.get(id); //지정된 id에 연결된 값을 반환
+        }
+        return null;
+    }
+
+    /**
+     * 조건을 만족하는 멤버들을 반환하는 메서드
+     */
+    @Override
+    public List<Member> findMembers(Predicate<Member> filter) {
+        return map.values().stream() //스트림 생성
+                .filter(filter) //필터링 조건 적용
+                .collect(Collectors.toList()); //결과를 리스트로 수집하여 반환
+
+    }
+
+    /**
+     * 조건을 만족하는 멤버 하나를 반환하는 메서드,
+     * 조건을 만족하는 멤버가 없을 수도 있으니 Optional을 반환
+     */
+    @Override
+    public Optional<Member> findMember(Predicate<Member> filter) {
+        return map.values().stream() //스트림 생성
+                .filter(filter) //필터링 조건 적용
+                .findAny(); //멤버 하나 선택하여 Optional로 감싸서 반환, 조건에 맞는 요소가 없다면 Optional,empty() 반환
+    }
+
+    /**
+     * 모든 멤버를 출력하는 메서드
+     */
+    @Override
+    public void printMembers() {
+        for(Member member : map.values()){ //for-each 문으로 임시변수 member에 map.values()를 대입하여 출력
+            System.out.println(member);
+        }
+    }
```

Comment:
```suggestion
    @Override
    public void printMembers() {
        member.values()
                .forEach(m -> System.out.print(m.getName() + " "));
    }
}
```
`map` 메서드 중에 `forEach` 메서드도 있으니 알아두시면 좋을 것 같아요~